src/DataStructures/Tensor/EagerMath/Norms.hpp

```diff
@@ -0,0 +1,213 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/Numeric.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+namespace L2Norm_detail {
+template <typename DataType, typename Symm, typename IndexList>
+Scalar<DataType> pointwise_l2_norm_square(
+    const Tensor<DataType, Symm, IndexList>& tensor) noexcept {
+  auto pointwise_l2_normsq = make_with_value<Scalar<DataType>>(tensor, 0.);
+  for (auto tensor_element = tensor.begin(); tensor_element != tensor.end();
+       ++tensor_element) {
+    // In order to handle tensory symmetries, we multiply the square of each
+    // stored component with its multiplicity
+    get(pointwise_l2_normsq) +=
+        tensor.multiplicity(tensor_element) * square(*tensor_element);
+  }
+  return pointwise_l2_normsq;
+}
+}  // namespace L2Norm_detail
+
+// @{
+/*!
+ * \ingroup TensorGroup
+ * \brief Compute point-wise Frobenius norm of arbitrary Tensors.
```

Comment:
I have a mild preference for changing the brief to the more accessible "Compute point-wise Euclidean L2 norm of arbitrary Tensors". I do not feel strongly about this suggestion, so let me know if you would rather keep as it is. I am happy say "Frobenius" in the details below, as is currently worded.

Reply:
done